### DISCUSSION TITLE
How to enable ssl configuration?

### DISCUSSION THREAD

**Discussion by lucian-cm** [ArcadeData/arcadedb#751] [Q&A]
I love the the multi model database concept that arcadedb is providing.
I am having difficulties setting up Arcadedb to run in embedded mode (https://docs.arcadedb.com/#Embedded-Server) using SSL configuration.

I am using _com.arcadedb:arcadedb-server:22.11.1_
I tried configuring ssl.enabled to _true_ and also provided the rest of the ssl properties documented in the official docs at https://docs.arcadedb.com/#Settings
(ssl.keyStore, ssl.keyStorePass, ssl.trustStore, ssl.trustStorePass) but still Undertow is starting just on HTTP listener only.
![image](https://user-images.githubusercontent.com/3694615/211336705-736a454e-484c-4470-9e26-ada2df34456e.png)

Tried starting the Arcadedb server using code similar to below:
```
ContextConfiguration embeddedArcadeDbServerConfig = new ContextConfiguration();

/**
set up SERVER_ROOT_PATH, SERVER_DATABASE_DIRECTORY, SERVER_DATABASE_LOADATSTARTUP, BUCKET_DEFAULT_PAGE_SIZE  and HA_ENABLED keys in a very simple manner (non ha with local file system) and database can be used and is created on the file system.
**/
embeddedArcadeDbServerConfig.setValue("ssl.enabled", Boolean.TRUE);
embeddedArcadeDbServerConfig.setValue("ssl.trustStore", "/path/to/truststore"); // JKS truststore
embeddedArcadeDbServerConfig.setValue("ssl.truststorePass", "truststore_password_cleartext");
embeddedArcadeDbServerConfig.setValue("ssl.keyStore", "/path/to/keystore"); // JKS keystore
embeddedArcadeDbServerConfig.setValue("ssl.keyStorePass", "keystore_password_cleartext");

ArcadeDBServer arcadedbServer = new ArcadeDBServer(embeddedArcadeDbServerConfig);
arcadedbServer.start();

```
only http listener is added for Undertow irrespective of ssl.enabled=true or ssl.enabled=false
![image](https://user-images.githubusercontent.com/3694615/211332469-6fb2078b-cf6c-413d-b879-ce64fbda890e.png)

Can you provide some guidance on how to achieve this use case?

**Comment by lvca** [comment]
I checked around the whole codebase and the ArcadeDB settings for SSL are not used by the server, but only by clients when they open a connection. I've created an issue for supporting SSL. It should be straightforward thanks to Undertow.

https://github.com/ArcadeData/arcadedb/issues/775

**Reply by lvca** [reply]
Sorry it took so long

**Reply by lucian-cm** [reply]
@Ivca, thanks for following up on the matter.

**Reply by lvca** [reply]
So now should work, please let me know if you're still working with Arcade

**Reply by lucian-cm** [reply]
Thanks for the fix :) 
I am still looking into arcadedb
